---
title: "Adult vs. Fetal Prefrontal Cortex Differential Gene Expression and Epigenetic Roadmap Correlation"
author: "M. D'Amour"
date: "2/12/2017"
output: pdf_document
fontsize: 10.5pt
geometry: margin=0.8in
---

## Project Overview

![Adult vs Fetal Brain Differentially Expressed Genes](/Users/mikedamour/Genomics/gdsCap/gdsCapData/readsData/expGenes/VolcanoPlotDEGenes.png)

This project involves the analysis of RNA-seq data for twelve (12) samples of post-mortem dorso-lateral prefrontal cortex tissue taken from six (6) second-trimester fetuses and six (6) adults of age ~40.  The purpose of the project is to determine whether which genes are differentially expressed in brain tissue from fetus and from adults.  This data can be used then to research which genes are responsible for e.g. early brain development vs adult brain maintenance.  Correlation with the Epigenomic Roadmap data on H3K4me3 marks that may assist in this differential expression is also checked.

This report is designed to be fully reproducible in methods, code, and data.  It may be accessed in Github repository mikedamour/CortexDEGenes as Linux command line scripts and Rmarkdown with all code, data, methods, and program versions.

## Brain Tissue RNA-seq Data Access

Data generated by Jaffe *et al.*^1^ was used and may be accessed at the NCBI SRA repository URL https://www.ncbi.nlm.nih.gov/sra/?term=SRXxxxxxx using the Experiment numbers in the table below.

Each sample was sequenced twice on Illumina HiSeq 2000 equipment, post-processed with Illumina software, providing twenty-four (24) libraries of paired-end reads in .fastq format. (1.03TB in .fastq and 328GB in post-aligned .bam.)

A phenotype data table was compiled manually in Excel from meta data provided on the NCBI website for each of the SRX samples and SRR files.  Reads data was added to the table after alignment for later statistical analysis.  The data, in the form of the table below, was saved from Excel in tab-delimited text format (.txt) for later load into R environment.

| Sample | Cohort | Experiment | File | Age | RIN | Sex | Race | Replct | rds_map'd | map_% |
|:------:|:------:|:----------:|:----:|----:|:---:|:---:|:----:|:------:|----------:|:-----:|
| R3452 | fetal | SRX683795 | SRR1554537 | -0.38 | 9.6 | F | AA | 1 | 129220081 | 98.8% |
| R3452 | fetal | SRX683795 | SRR2071348 | -0.38 | 9.6 | F | AA | 2 | 243191549 | 90.7% |
| R3462 | fetal | SRX683796 | SRR1554538 | -0.40 | 6.4 | F | AA | 1 | 156236147 | 98.9% |
| R3462 | fetal | SRX683796 | SRR2071349 | -0.40 | 6.4 | F | AA | 2 | 462286753 | 92.9% |
| R3485 | fetal | SRX683799 | SRR1554541 | -0.38 | 5.7 | M | AA | 1 | 171773460 | 98.9% |
| R3485 | fetal | SRX683799 | SRR2071352 | -0.38 | 5.7 | M | AA | 2 | 192826059 | 91.7% |
| R4706 | fetal | SRX683824 | SRR1554566 | -0.50 | 8.3 | M | HISP | 1 | 123431458 | 98.7% |
| R4706 | fetal | SRX683824 | SRR2071377 | -0.50 | 8.3 | M | HISP | 2 | 134015735 | 94.0% |
| R4707 | fetal | SRX683825 | SRR1554567 | -0.40 | 8.6 | M | AA | 1 | 143917249 | 98.9% |
| R4707 | fetal | SRX683825 | SRR2071378 | -0.40 | 8.6 | M | AA | 2 | 155804075 | 93.9% |
| R4708 | fetal | SRX683826 | SRR1554568 | -0.50 | 8.0 | M | AA | 1 | 113018703 | 98.9% |
| R4708 | fetal | SRX683826 | SRR2071379 | -0.50 | 8.0 | M | AA | 2 | 233805731 | 91.7% |
| R2869 | adult | SRX683793 | SRR1554535 | 41.58 | 8.7 | M | AA | 1 | 96258319 | 99.1% |
| R2869 | adult | SRX683793 | SRR2071346 | 41.58 | 8.7 | M | AA | 2 | 118061708 | 81.6% |
| R3098 | adult | SRX683794 | SRR1554536 | 44.17 | 5.3 | F | AA | 1 | 49744953 | 99.5% |
| R3098 | adult | SRX683794 | SRR2071347 | 44.17 | 5.3 | F | AA | 2 | 77925728 | 92.5% |
| R3467 | adult | SRX683797 | SRR1554539 | 36.50 | 9.0 | F | AA | 1 | 80649750 | 99.2% |
| R3467 | adult | SRX683797 | SRR2071350 | 36.50 | 9.0 | F | AA | 2 | 88041049 | 83.8% |
| R3969 | adult | SRX683814 | SRR1554556 | 36.98 | 8.5 | M | AA | 1 | 113119523 | 99.2% |
| R3969 | adult | SRX683814 | SRR2071367 | 36.98 | 8.5 | M | AA | 2 | 123291026 | 90.6% |
| R4166 | adult | SRX683819 | SRR1554561 | 43.88 | 8.7 | M | AA | 1 | 93639705 | 99.0% |
| R4166 | adult | SRX683819 | SRR2071372 | 43.88 | 8.7 | M | AA | 2 | 106177991 | 84.6% |
| R3969 | adult | SRX683792 | SRR1554534 | 40.42 | 8.4 | M | AA | 1 | 67363924 | 98.9% |
| R3969 | adult | SRX683792 | SRR2071345 | 40.42 | 8.4 | M | AA | 2 | 77828716 | 85.2% | 

## Hisat2 Sequence Alignment, QC, and Stats at Linux Command Line

Hisat2 was used to align the reads to the *H. sapiens* UCSC GRCh38 human genome (hg38).  The publishers of Hisat2 make available at https://ccb.jhu.edu/software/hisat2/index.shtml a pre-indexed version of hg38 downloadable by web interface.  The SRA reads files were accessed directly using the Hisat2 SRA interface option.  

Hisat2, compression to .bam, FastQC, and stats analysis were run at the command line from the following scripts.

```{r setup, eval=FALSE}
GENEDIR=/pathToGenomeData
ARCVDIR=/pathToArchive
LOGDIR=/pathToLog
WRKDIR=/pathToWorkDir

# Create file listing all the SRR numbers for files that are to be analyzed
printf '%s\n' 'SRR1554534' 'SRR1554535' 'SRR1554536' 'SRR1554537' 'SRR1554538' 'SRR1554539' \
'SRR1554541' 'SRR1554556' 'SRR1554561' 'SRR1554566' 'SRR1554567' 'SRR1554568' 'SRR2071345' \
'SRR2071346' 'SRR2071347' 'SRR2071348' 'SRR2071349' 'SRR2071350' 'SRR2071352' 'SRR2071367' \
'SRR2071372' 'SRR2071377' 'SRR2071378' 'SRR2071379' >srrList.txt

# Hisat2 only outputs .sam files, so separately compress to .bam
for i in `cat srrList.txt`; do
    hisat2 -p 4 -t -x $GENEDIR/hg38_ht/genome --sra-acc i -S $ARCVDIR/i.sam >& $LOGDIR/i.log
    samtools view -b -@ 4 -o $WRKDIR/i.bam $ARCVDIR/i.sam
    # Check quality of Hisat2 output .bam files
    fastqc -o $WRKDIR/fastqcOut -t 4 $WRKDIR/i.bam
    samtools stats $WRKDIR/i.bam > $WRKDIR/statsOut/i.stats
done
```
The (example) lines below were extracted from each .stats file for reporting. The "reads mapped" values were manually inserted into the phenotype table for later FPMR calculation for plotting.  
        # SN	raw total sequences:	132911310  
        # SN	reads mapped:	        106284374  
        # SN	average quality:	    27.9  

## Review of QC Data

The QC data output by FastQC above was analyzed using MultiQC.  Directing MultiQC to the main working directory, it locates subdirectories containing FastQC output and logs, consolidating them into .html and text output.

```{r, eval=FALSE}
# Run MultiQC on all QC and log data
multiqc -f  $WRKDIR/readsData
```

After review of the MultiQC .html output file and, since this analysis is for gene expression read counts, not for SNPs, no further trimming was done for base quality, read quality, or contamination.

## Counting Reads for Gene Expression

The Gencode release 25 GCRh38.p7 in .gtf format was downloaded from https://www.gencodegenes.org/releases/current.html using the web interface. The featureCount program was run at the command line, as follows.

```{r, eval=FALSE}
featureCounts -T 4 -t gene -p -a $GENES/hg38genes/gencode.v25.annotation.gtf \
-o $WKDIR/expGenes/aln.pg.fcnt \
$WRKDIR/SRR1554535.bam $WRKDIR/SRR2071346.bam $WRKDIR/SRR1554536.bam \
$WRKDIR/SRR2071347.bam $WRKDIR/SRR1554539.bam $WRKDIR/SRR2071350.bam \
$WRKDIR/SRR1554556.bam $WRKDIR/SRR2071367.bam $WRKDIR/SRR1554561.bam \
$WRKDIR/SRR2071372.bam $WRKDIR/SRR1554534.bam $WRKDIR/SRR2071345.bam \
$WRKDIR/SRR1554537.bam $WRKDIR/SRR2071348.bam $WRKDIR/SRR1554538.bam \
$WRKDIR/SRR2071349.bam $WRKDIR/SRR1554541.bam $WRKDIR/SRR2071352.bam \
$WRKDIR/SRR1554566.bam $WRKDIR/SRR2071377.bam $WRKDIR/SRR1554567.bam \
$WRKDIR/SRR2071378.bam $WRKDIR/SRR1554568.bam $WRKDIR/SRR2071379.bam
```

## Exploratory Analysis of Gene Expression

From this point, all analysis was performed in R.  Initializing environment.
```{r, eval=TRUE, message=FALSE}
library(Rsubread);  library(GenomicRanges);   library(BiocGenerics)
library(DESeq2);    library(AnnotationDbi);   library(SummarizedExperiment)
library(Biobase);   library(stringr);         library(ChIPpeakAnno)
library(dplyr);     library(RColorBrewer);    library(AnnotationHub)
library(pheatmap);  library(rtracklayer);     library(ggplot2);   library(ggrepel)

```
## Data Preparation - RangedSummarizedExperiment Object

Followed RNA-Seq workflow published by Love *et al.*^2^
```{r, include= TRUE, cache=TRUE, cache.comments=TRUE}
bamDir = c('/Volumes/MacExp/geneData/hisatBam/')
rsltDir = c('/Users/mikedamour/Genomics/gdsCap/gdsCapData/readsData/expGenes/')
geneDir = c('/Users/mikedamour/Genomics/genomes/hg38genes/')

# Import data from command line featureCounts
expGenesCL = read.table (paste0(rsltDir, 'aln.pg.fcnt.txt'), 
                         header = TRUE, sep = '\t', skip = 1)

# Clean col names to include only the file id
colnames(expGenesCL) = c(colnames(expGenesCL[1:6]), 
                         str_extract(colnames(expGenesCL[7:30]), "SRR[0-9]+"))
```
```{r, include=TRUE, eval=FALSE}
# Pull gene symbols out of the GTF file used for featureCounts - takes about 3 minutes
# Write into a file first time.  Read from file after.
hg38GeneSymsRaw = read.table(paste0(geneDir, "gencode.v25.annotation.gtf"), 
                             fill = TRUE, skip = 5, stringsAsFactors = TRUE)
hg38GeneSyms = select(geneSymsRaw[hg38GeneSymsRaw$V3 == "gene",], V10, V19)
row.names(hg38GeneSyms) = NULL; colnames(hg38GeneSyms) = c("ENSEMBL", "SYMBOL")
write.table(hg38GeneSyms, paste0(geneDir, "hg38GeneSyms.txt"), quote = FALSE)
```
```{r, include=TRUE, eval=TRUE, cache=TRUE, cache.comments=TRUE}
# Recover previously written gene symbol file
hg38GeneSyms = read.table(paste0(geneDir, "hg38GeneSyms.txt"), stringsAsFactors = FALSE)
expGenesCL$hg38Sym = hg38GeneSyms$SYMBOL[match(expGenesCL[ ,1], hg38GeneSyms$ENSEMBL)]
expGenesCL = select(expGenesCL, Geneid, hg38Sym, Chr:SRR2071379)
dim(expGenesCL)  # 58037  31  # All the genes and 31 columns
write.table(expGenesCL, paste0(rsltDir, "dlpfcDEGFull.txt"), sep = '\t', quote = FALSE)

# Extract assay information and row name
expData = DataFrame(expGenesCL[,8:31])
row.names(expData) = expGenesCL$Geneid

# Make a GRanges for the row ranges
expGr = makeGRangesFromDataFrame(expGenesCL[,1:6], keep.extra.columns = TRUE)

# Bring in phenotype data
pdata = read.table (paste0(rsltDir, 'phenoData2_1-18-17.txt'), header = TRUE, sep = '\t')
pdata = pdata[1:24,]
pdata$Sample = as.character(pdata$Sample)
pdata$Sample = as.factor(str_extract(pdata$Sample, "R[0-9]+"))
pdata$Replicate = as.factor(pdata$Replicate)
row.names(pdata) = pdata$File

# Construct the Ranged SE
pfcSe = SummarizedExperiment(assays = list(counts = as.matrix(expData)), 
                             rowRanges = expGr, colData = pdata)
# Use adult as the reference level for the cohort variable
pfcSe$Cohort = relevel(pfcSe$Cohort, 'adult')  # Already alphabetic, but to be sure
```

## Begin Transform of Data for Visual Exploration

```{r, eval=TRUE, cache=TRUE, cache.comments=TRUE}
nrow(pfcSe)  # 58037
pfcSe = pfcSe[rowSums(assay(pfcSe)) > 1,]  # Remove empty rows and chrM
pfcSe = subset(pfcSe, seqnames != 'chrM')
nrow(pfcSe)  # 32683

# Make a collapsed sample SE with values as fragments per million reads, not FPKM
pfcSeColl = collapseReplicates(pfcSe, pfcSe$Sample, renameCols = TRUE)
# Make an fpmr assay for plotting
fpmrAssay = log2(assay(pfcSeColl) / (pfcSeColl$Reads_Mapped/1e6) + 1)

# Cannot convert se to dds due to bug? Ensure pfcSE built above with matrix not DataFrame
pfcDds = DESeqDataSet(pfcSe, ~ RIN + Sex + Replicate + Race + Dup_Lev + Contam + Cohort)
# Assay must be integers, so must wait to collapseReplicates and any FPMR stuff
pfcDdsColl = DESeqDataSet(pfcSeColl, ~ RIN + Sex + Race + Dup_Lev + Contam + Cohort)
pfcDdsColl$Replicate = droplevels(pfcDdsColl$Replicate)  # After subset columns

# Reg log transform
pfcSeRL = rlog(pfcDds, blind = FALSE)  # Keep replicates separate for PCA
pfcSeCollRL = rlog(pfcDdsColl, blind = FALSE)  # Replicates collapsed for boxplot
```

## Plot Sample Data to Explore Transform

```{r, eval=TRUE}
par(mfrow = c(1,2))
boxplot(fpmrAssay, col = "yellow", main = "Gene Counts FPMR log2",
        ylab = "log2 FPMR Gene Count", cex.axis = 0.6, cex.lab = 0.8)
boxplot(assay(pfcSeCollRL), col = "lightgreen", main = "Gene Counts rlog Transform",
        ylab = "rlog Gene Counts",cex.axis = 0.6, cex.lab = 0.8)
```

## Plot Primary Component Analysis

The first plot highlights Cohort vs. Sex and in fact shows that these variables represent the primary and secondary component of variance.  Jaffe mentioned in his lecture that RIN was an important component of variance.  PCA plot does not bear this out.  Suspicion of artificats in the technical replicates was plotted and proved to be unfounded.  Note that one fetal sample was assigned sex of male when all data seem to indicate - including the chrY expression data - that the sample is female.  See the bottom right corner of the first PCA plot.
```{r, eval=TRUE}
plotPCA(pfcSeCollRL, intgroup = c("Cohort", "Sex"))
plotPCA(pfcSeCollRL, intgroup = c("Cohort", "RIN"))
plotPCA(pfcSeRL, intgroup = c("Cohort", "Replicate"))
plotPCA(pfcSeRL, intgroup = c("Sex", "Replicate"))
```

## Statistical Analysis of Expressed Genes

This stage of analysis will determine statistically significant differential gene expression between adult and fetal brain.
```{r, eval=TRUE, cache=TRUE, cache.comments=TRUE, message=FALSE}
# Simplify formula and run analysis
ddsRS = DESeqDataSet(pfcSeColl, ~ RIN + Sex + Cohort)
ddsRS = DESeq(ddsRS)

# Extract results and annotate with gene symbols
resRS = results(ddsRS, alpha = 0.01, contrast = c("Cohort", "fetal", "adult"))
resRS$hg38Sym = hg38GeneSyms$SYMBOL[match(row.names(resRS), hg38GeneSyms$ENSEMBL)]
```

## Volcano Plot Differential Gene Expression Fetal vs Adult(ref) Brain

After Turner^3^.
```{r, eval=TRUE, include=TRUE, message=FALSE, error=FALSE}
# Add results data to DESeq object for tidy
mcols(ddsRS) = as(mutate(as.data.frame(mcols(ddsRS)), FvAl2fc = resRS$log2FoldChange,
                         FvApvalue = resRS$pvalue,FvApadj = resRS$padj), "DataFrame")
resRSdf = mutate(as.data.frame(resRS),
                 sig = ifelse((resRS$padj<10e-120) |
                                (resRS$log2FoldChange > 8.5) |
                                (resRS$log2FoldChange < -10),
                              "High Signif", "Less Signif"))
p = ggplot(resRSdf, aes(log2FoldChange, -log10(pvalue))) +
  geom_point(aes(col=sig)) +
  scale_color_manual(values=c("red", "lightblue3")) +
  geom_text_repel(data=filter(resRSdf, sig == "High Signif"), aes(label=hg38Sym))
p
```

## Epigenome Roadmap Data  

The H3K4me3 peak data was identified at the Epigeneome Roadmap website with URL given below.  Files were loaded through Bioconductor AnnotationHub already in R GRanges formats.

All gene expression data was aligned to the hg38 genome but the Epigenome Roadmap data was aligned to hg19.  This required doing liftOver from hg19 to hg38 to proceed.

```{r, eval=TRUE, cache=TRUE, message=FALSE, error=FALSE, warning=FALSE}
# Roadmap data references at http://egg2.wustl.edu/roadmap/web_portal/meta.html
# Go to Annotation hub for data - all hg19!  Load liftOver chain file.
loChain = import.chain(paste0(geneDir, "hg19ToHg38.over.chain"))
ahub = AnnotationHub()
ahub = subset(ahub, species == "Homo sapiens")

# Get roadmap data and liftover to hg38 from hg19, start with fetal brain data
qh = query(ahub, c('h3k4me3', 'Homo sapiens', 'narrowPeak', 'brain', 'E082'))
fb19 = qh[['AH30479']]  # Female fetal E082
fb = liftOver(fb19, loChain); fb = unlist(fb)

# Adult brain data
qh = AnnotationHub::query(ahub, c('h3k4me3', 'Homo sapiens', 'narrowPeak', 'brain', 'E073'))
ab19 = qh[['AH30413']]  # Male mixed adult E073
ab = liftOver(ab19, loChain); ab = unlist(ab)

# Adult liver data
qh = query(ahub, c('h3k4me3', 'Homo sapiens', 'narrowPeak', 'liver', 'E066'))
al19 = qh[['AH30367']]  # Adult liver E066
al = liftOver(al19, loChain); al = unlist(al)
```
## Analysis of Brain Gene Expression Overlaps

The brain tissue differential expression data is checked for overlap with fetal brain, adult brain and liver H3K4me3 peaks. Only DE genes above FDR 0.01 and log2 fold change > 1 were considered.  

```{r, message=TRUE, eval=TRUE, fig.height=3, fig.width=4, fig.align='center'}
# Compare to highly expressed genes
fdr = 0.01; l2fc = 1
ddsRSSig = subset(ddsRS, FvApadj < fdr & abs(FvAl2fc) > l2fc)
fbOlap = subsetByOverlaps(promoters(rowRanges(ddsRSSig)), fb)
abOlap = subsetByOverlaps(promoters(rowRanges(ddsRSSig)), ab)
alOlap = subsetByOverlaps(promoters(rowRanges(ddsRSSig)), al)

# Plot Venn diagram of fetal brain vs adult brain vs adult liver
par(mfrow=c(1,1))
makeVennDiagram(list(fbOlap, abOlap, alOlap), ignore.strand = T,
                NameOfPeaks = c('Fetal Brain', 'Adult Brain', 'Adult Liver'),
                totalTest=10000, scaled=F, euler.d=F, fill=c(1,2,3))
```
Code related to Hansen^4^.
```{r, eval=TRUE}
# Check for H3K4me3 mark enrichment in fetal brain vs. adult brain vs. adult liver
methylation = data.frame(MethOlaps = c("FetalBrainOlaps", "AdultBrainOlaps", "AdultLiverOlaps"), 
                         OddsRatio = c(0,0,0))
refPeak = c('fb', 'ab', 'al')
prom = reduce(promoters(rowRanges(ddsRSSig), ignore.strand = TRUE))
for(i in seq_along(refPeak))  { 
  peaks = reduce(get(refPeak[i]))
  both <- GenomicRanges::intersect(prom, peaks, ignore.strand = TRUE)
  only.prom <- BiocGenerics::setdiff(prom, both)
  only.peaks <- BiocGenerics::setdiff(peaks, both)
  overlapMat <- matrix(0, ncol = 2, nrow = 2)
  colnames(overlapMat) <- c("in.peaks", "out.peaks")
  rownames(overlapMat) <- c("in.promoters", "out.promoter")
  overlapMat[1,1] <- sum(width(both))
  overlapMat[1,2] <- sum(width(only.prom))
  overlapMat[2,1] <- sum(width(only.peaks))
  overlapMat[2,2] <- 1.5*(10^9) - sum(overlapMat)
  round(overlapMat / 10^6, 2)
  
  print(as.character(methylation$MethOlaps[i]))
  print(round(overlapMat/ 10^6, 1))
  print("---------------------------")
  oddsRatio <- overlapMat[1,1] * overlapMat[2,2] / (overlapMat[2,1] * overlapMat[1,2])
  methylation$OddsRatio[i] <-oddsRatio
}
# Show odds ratios for fetal brain, adult brain, and adult liver overlaps with DE genes
print(methylation)
```

The table above shows the odds ratio of overlaps with highly expressed genes.  Notice that the odds ratio for adult liver methylation marks with the fetal/adult DE genes is significantly lower.

## Conclusion

### Promoter Comparison - Fetal Brain and Adult Brain

There are significant differences in H3K4me3 marks between fetal and adult brain as seen with the Venn diagram and contingency table.  The overlap with brain methylation peaks are significantly different between fetal and adult samples, with fetal brain having 205 highly expressed genes promoters marked H3K4me3 completely separately from adult brain, with adult brain having 526 such separately marked genes.

### Sanity Check

There are significantly fewer methylation peaks from the liver ChIPseq data that overlap with either the fetal brain or adult brain differentially expressed genes, as shown by the tables and diagram above.

### References
1. Jaffe, A.E., Shin, J., Collado-Torres, L., Leek, J.T., Tao, R., Li, C., Gao, Y., Jia, Y., Maher, B.J., Hydel, T.M., Kleinman, J.E., Weinberger, D.R.  Developmental regulation of human cortex transcription and its clinical relevance at single base resolution. *Nature Neuroscience* **18**, 154-161 (2015)
2. Love, M., Anders, S., Huber, W., Morgan M.  RNA-Seq workflow: gene-level exploratory analysis and differential expression.  *Bioconductor* http://www.bioconductor.org/help/workflows/rnaseqGene/ (2017)
3. Turner, S. *Getting Genetics Done*, http://www.gettinggeneticsdone.com/2014. 
4. Hansen, K.D. Usecase - Basic GRanges and AnnotationHub. http://kasperdanielhansen.github.io/genbioconductor/html/Usecase_AnnotationHub_GRanges.html

```{r}
sessionInfo()
```
